src/contract.rs

```diff
@@ -506,13 +573,13 @@ fn try_redeem<S: Storage, A: Api, Q: Querier>(
         )));
     }
 
-    let mut config = Config::from_storage(&mut deps.storage);
-    let total_supply = config.total_supply();
-    if let Some(total_supply) = total_supply.checked_sub(amount_raw) {
-        config.set_total_supply(total_supply);
-    } else {
+    let token_reserve = deps
+        .querier
+        .query_balance(&env.contract.address, "uscrt")?
+        .amount;
+    if amount > token_reserve {
         return Err(StdError::generic_err(
-            "You are tyring to redeem more tokens than what is available in the total supply",
+            "You are trying to redeem for more SCRT than the token has in its deposit reserve.",
         ));
```

Comment:
I think this is not necessary in the reference implementation. If we passed the internal consistency check against `config.total_supply()` then there's no way for `amount > token_reserve` to be true. In other implementations that have other ways of pulling SCRT out of the contract, this may be different but this impl is just wasting gas with this check.

Reply:
It would be possible if someone set a config where deposit is disabled and redeem is enabled.  While I wouldn't know exactly WHY someone would do it, it is possible that for some reason they don't want people minting tokens through a deposit, and the app using the snip20 actually loads the contract with SCRT for people to redeem against

Reply:
ahhh i see what you mean. This does make sense. For sSCRT (and in the future things like IBC sATOM, sLUNA, etc) we can just cut out that check, but it's a good check to have for general purpose usage.